eth/downloader/downloader.go

```diff
@@ -587,41 +587,87 @@ func (d *Downloader) fetchHeight(p *peerConnection) (*types.Header, error) {
 	}
 }
 
+// calcRequestSpan calculates what headers to request from a peer when trying to determine the
+// common ancestor.
+func calculateRequestSpan(remoteHeight, localHeight uint64) (int64, int, int, uint64) {
+	var (
+		from     int
+		count    int
+		MaxCount = MaxHeaderFetch / 16
+	)
+	// requestHead is the highest block that we will ask for. If requestHead is not offset,
+	// the highest block that we will get is 16 blocks back from head, which means we
+	// will fetch 14 or 15 blocks unneccesarily in the case the height difference
+	// between us and the peer is 1-2 blocks, which is most common
+	requestHead := int(remoteHeight) - 1
+	if requestHead < 0 {
+		requestHead = 0
+	}
+	// requestBottom is the lowest block we want included in the query
+	// Ideally, we want to include just below own head
+	requestBottom := int(localHeight - 1)
+	if requestBottom < 0 {
+		requestBottom = 0
+	}
+	totalSpan := requestHead - requestBottom
```

Comment:
What happens here if our local height is larger than the remote one? Total Span will be negative. Is that acceptable?

Reply:
That works, although I agree it's not obvious. But it's in the testcases:
```
		{1000, 1500,
			[]int{997, 999},
```
So remote `1000`, local `1500`, will request `997, 998`.